liblangutil/SemVerHandler.cpp

```diff
@@ -265,14 +272,16 @@ unsigned SemVerMatchExpressionParser::parseVersionPart()
 		{
 			c = currentChar();
 			if (v * 10 < v || v * 10 + static_cast<unsigned>(c - '0') < v * 10)
-				BOOST_THROW_EXCEPTION(SemVerError());
+				BOOST_THROW_EXCEPTION(SemVerError() << errinfo_comment("Invalid token in version pragma: "s + c));
```

Comment:
```suggestion
				BOOST_THROW_EXCEPTION(SemVerError() << errinfo_comment("Integer too large to be used in a version number."));
```
You might also include the current value of `v`, though that's not the complete integer so it might be misleading...

Reply:
Also, there seems to be no test case covering this.